subsys/bluetooth/services/fast_pair/fp_auth.c

```diff
@@ -245,6 +212,20 @@ int fp_auth_start(struct bt_conn *conn, bool send_pairing_req)
 	bt_auth_peer_passkeys[conn_idx] = EMPTY_PASSKEY;
 	update_conn_handling(conn, true);
 
+	err = bt_conn_auth_cb_overlay(conn, &conn_auth_callbacks);
+	if (err) {
+		return err;
+	}
```

Comment:
Wouldn't it be better to overlay the callbacks first and then update the conn handling flag. This way, you will filter pairing attempts when the Fast Pair module isn't yet ready.

Reply:
That seems better approach. Changed. Thanks for the comment